Firestore/core/src/firebase/firestore/util/path.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_PATH_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_PATH_H_
+
+#include <string>
+#include <utility>
+
+#include "absl/strings/string_view.h"
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+struct Path {
```

Comment:
My intuition is that:
  * Names like `Join` should be qualified somehow, though `Dirname` could be free.
  * It's likely that you want more than one of these at a time.
  * It's simple and easy to get them all with a single `using` declaration.
  * The alternative requires doing something like `namespace path = firebase::firestore::util::path`

Basically, I see these as a tight knit group that don't make much sense individually and the extra qualifier makes it more obvious what we're talking about, improving the names. Another example that looks like this already is `OrderedCode`.

I can see it working the other way too, and like you don't have strong feelings about it.

I'm starting to feel like the `util` namespace is getting somewhat full, so maybe it would make sense to move these out.

Reply:
Makes sense, thanks for the explanation.